packages/flutter/lib/src/painting/box_painter.dart

```diff
@@ -679,6 +682,46 @@ class LinearGradient extends Gradient {
     );
   }
 
+  /// Returns a new [LinearGradient] with its properties scaled by the given factor.
+  LinearGradient scale(double factor) {
+    return new LinearGradient(
+      begin: begin,
+      end: end,
+      colors: colors.map<Color>((Color color) => Color.lerp(null, color, factor)).toList(),
+      stops: stops,
+      tileMode: tileMode,
+    );
+  }
+
+  /// Linearly interpolate between two [LinearGradient]s.
+  ///
+  /// If either gradient is null, this function linearly interpolates from a
+  /// a gradient that matches the other gradient in begin, end, stops and 
+  /// tileMode and with the same colors but transparent.
+  static LinearGradient lerp(LinearGradient a, LinearGradient b, double t) {
+    if (a == null && b == null)
+      return null;
+    if (a == null)
+      return b.scale(t);
+    if (b == null)
+      return a.scale(1.0 - t);
+    // Interpolation is only possible when the lengths of colors and stops are
+    // the same or stops is null for one.
+    assert(a.colors.length == b.colors.length);
```

Comment:
one could imagine doing a lerp where we add stops... maybe turn the comment here into a TODO since eventually we'll want to fix it.